src/tir/transforms/split_host_device.cc

```diff
@@ -104,44 +81,42 @@ class VarUseDefAnalysis : public StmtExprMutator {
       dyn_shmem_size_ = dyn_shmem_size_ * (op->dtype.bytes());
       use_dyn_shmem_ = true;
     }
-    return StmtExprMutator::VisitStmt_(op);
+    StmtVisitor::VisitStmt_(op);
   }
 
-  Stmt VisitStmt_(const AllocateConstNode* op) final {
-    this->HandleDef(op->buffer_var.get());
-    return StmtExprMutator::VisitStmt_(op);
-  }
+  // recording what thread axis have been visited.
+  std::unordered_set<const IterVarNode*> defined_thread;
+};
 
-  Stmt VisitStmt_(const StoreNode* op) final {
-    LOG(FATAL) << "Unexpected use of deprecated StoreNode.  Please use BufferStoreNode instead.";
-  }
+/*!
+ * \brief Mutator class to remove unrefenced let stmt/expressions.
+ * \param use_count The pre-computed variable to use count map.
+ */
+class UnreferencedLetRemover : public StmtExprMutator {
```

Comment:
Thank you for making this change, and it really helps in the readability to have the analysis/mutation be separated out!

Reply:
Thank you for your review! I have updated the code according to your suggestions.